omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/TransactionAspect.java

```diff
@@ -46,6 +48,20 @@ public TransactionAspect(SagaMessageSender sender, OmegaContext context) {
   @Around("execution(@org.apache.servicecomb.pack.omega.transaction.annotations.Compensable * *(..)) && @annotation(compensable)")
   Object advise(ProceedingJoinPoint joinPoint, Compensable compensable) throws Throwable {
     Method method = ((MethodSignature) joinPoint.getSignature()).getMethod();
+    // just check if we need to setup the transaction context information first
+    TransactionContext transactionContext = getTransactionContextFromArgs(joinPoint.getArgs());
+    if (transactionContext != null) {
+      if (context.globalTxId() != null) {
```

Comment:
If you first determine if context.globalTxId() is not empty, is it better?

Reply:
Current we just give a chance to let custom to override the transaction context setting as he want.  The check of globalTxId is just for warning the user about the side effect, if the user find out the GID is set, he could chose not to pass the parameter into the sub transaction method.

Reply:
Ok, this looks good.